### PR TITLE
contribution guide

### PR DESCRIPTION
**Pull request by twaugh**
Suggestions welcome.

### CODE REVIEW

**Comment by vrutkovs** [comment]
Lets rename this to `CONTRIBUTING.md` and move to the root of the repo so that Github shows this every time a new issue/PR is filed?

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+Contribution Guide
+==================
+
+Licensing
+---------
+
+This project is licensed using the [BSD-3-Clause license](https://github.com/projectatomic/osbs-client/blob/master/LICENSE). When submitting pull requests please make sure your commit messages include a signed-off-by line. You can do this by using `git commit --signoff`.
```
Comment (lcarva) [inline docs/contributing.md]: Maybe a link to this is helpful? https://help.github.com/articles/signing-commits-using-gpg/

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+Please create your feature branch from the *master* branch. Make sure to add unit tests under the `tests/` subdirectory (we use py.test and flexmock for this). Tests are run automatically when you push new commits, but you can run them locally with `py.test tests` from the top directory.
+
+Follow the PEP8 coding style.
+
+Before a pull request is approved it must meet these criteria:
+- unit tests pass
+- code coverage from testing  does not decrease and new code is covered
+
+Once it is approved by two developer team members it may be merged. To avoid creating merge commits the pull request will be rebased during the merge.
```
Comment (lcarva) [inline docs/contributing.md]: I think it still creates merge commits, but the commits are squashed.

Comment (lcarva) [inline docs/contributing.md]: How about making a note about the git commit message? Maybe a reference to this is sufficient? http://chris.beams.io/posts/git-commit/

File: docs/contributing.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+This project is licensed using the [BSD-3-Clause license](https://github.com/projectatomic/osbs-client/blob/master/LICENSE). When submitting pull requests please make sure your commit messages include a signed-off-by line. You can do this by using `git commit --signoff`.
+
+Submitting changes
+------------------
+
+Changes are accepted through pull requests.
+
+Please create your feature branch from the *master* branch. Make sure to add unit tests under the `tests/` subdirectory (we use py.test and flexmock for this). Tests are run automatically when you push new commits, but you can run them locally with `py.test tests` from the top directory.
+
+Follow the PEP8 coding style.
```
Comment (lcarva) [inline docs/contributing.md]: Call out that we allow 99 chars per line, instead of 79?

File: docs/contributing.md (hunk 2 above)
Comment (csomh) [inline docs/contributing.md]: `Rebase and merge` and `Squash and merge` are two separate options in GitHub, afaik. I we wan't both we might have to do the merges locally with git. Or simply asks the contributors to rebase and squash before the merge.

File: docs/contributing.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+---------
+
+This project is licensed using the [BSD-3-Clause license](https://github.com/projectatomic/osbs-client/blob/master/LICENSE). When submitting pull requests please make sure your commit messages include a signed-off-by line. You can do this by using `git commit --signoff`.
+
+Submitting changes
+------------------
+
+Changes are accepted through pull requests.
+
+Please create your feature branch from the *master* branch. Make sure to add unit tests under the `tests/` subdirectory (we use py.test and flexmock for this). Tests are run automatically when you push new commits, but you can run them locally with `py.test tests` from the top directory.
```
Comment (csomh) [inline docs/contributing.md]: When you push new commits, tests will be triggered to be run in [travis-ci](https://travis-ci.org/projectatomic/osbs-client) and results will be shown in the PR. You can also run them...

File: docs/contributing.md (hunk 2 above)
Comment (lcarva) [inline docs/contributing.md]: Btw, neither `Rebase and merge` and `Squash and merge` create a merge commit, which is much cleaner than `Merge pull request`.

**Pull request merged**